R/touchstone_demographic_dataset.R

```diff
@@ -0,0 +1,245 @@
+# Get all touchstone demographic dataset rows for a set
+# of touchstones. Also lookup both strings and numerical
+# ids for demographic source and statistic type.
+
+db_tdd_for_touchstones <- function(con, touchstones) {
+  DBI::dbGetQuery(con, sprintf("
+    SELECT touchstone_demographic_dataset.id, touchstone,
+           demographic_dataset,
+           demographic_source.code as dsource_code,
+           demographic_statistic_type.code as dtype_code,
+           demographic_source.id as dsource,
+           demographic_statistic_type.id as dtype
+      FROM touchstone_demographic_dataset
+      JOIN demographic_dataset
+        ON demographic_dataset.id =
+             touchstone_demographic_dataset.demographic_dataset
+      JOIN demographic_source
+        ON demographic_source.id = demographic_dataset.demographic_source
+      JOIN demographic_statistic_type
+        ON demographic_statistic_type.id =
+             demographic_dataset.demographic_statistic_type
+     WHERE touchstone IN %s", sql_in(touchstones)))
+}
+
+# Get all demographic_dataset information to cover the given
+# types and sources (which are character). The query is a bit
+# generous - lookup all types and sources, not just the specific
+# combinations of those that we want.
+
+db_dd_info <- function(con, types, sources) {
+  DBI::dbGetQuery(con, sprintf("
+    SELECT demographic_dataset.demographic_source,
+           demographic_dataset.demographic_statistic_type,
+           demographic_source.code AS dsource_code,
+           demographic_statistic_type.code AS dtype_code,
+           demographic_dataset.id
+      FROM demographic_dataset
+      JOIN demographic_source
+        ON demographic_dataset.demographic_source = demographic_source.id
+      JOIN demographic_statistic_type
+        ON demographic_dataset.demographic_statistic_type =
+             demographic_statistic_type.id
+     WHERE demographic_statistic_type.code IN %s
+        OR demographic_source.code IN %s",
+                             sql_in(unique(types)),
+                             sql_in(unique(sources))))
+}
+
+extract_touchstone_demographic_dataset <- function(e, path, con) {
+
+  # Do we have any touchstone_demographic_dataset work to do?
+
+  if (!is.null(e$touchstone_demographic_dataset_csv)) {
+
+    e_tdd_csv <- e$touchstone_demographic_dataset_csv
+
+    # Test csv columns are correct.
+
+    if (length(names(e_tdd_csv)) != 3) {
```

Comment:
you can use `setequal(names(x), expected)` to check length and content of a group of names without regard to their order

Reply:
Done - nice.